### PR TITLE
Feature/experimental refactor

### PR DESCRIPTION
**Pull request by mikeAdamss**
Merged `nop_cache`.

Attempting to decouple `token` and `identity` concepts.

### CODE REVIEW

File: cache/nop_cache.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		"key":   key,
+		"ID":    i.ID,
+		"email": i.Email,
+		"ttl":   ttl.Seconds(),
+	})
+	return nil
+}
+
+// StoreToken .... caches a token
+func (c *NOPCache) StoreToken(token string, identityID string) (time.Duration, error) {
```
Comment (daiLlew) [inline cache/nop_cache.go:24]: I am not sure what the difference is between `SetToken` and `StoreToken`?

File: cache/nop_cache.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	log.Info("nopcache: store token", log.Data{"key": token})
+	return 0, nil
+}
+
+// GetToken .... gets a token from the cache
+func (c *NOPCache) GetToken(token string, identityID string) (time.Duration, error) {
```
Comment (daiLlew) [inline cache/nop_cache.go:30]: Im not sure we need to pass in the identity here - can't remember what we discussed in the meeting? Im thinking we pass in the token and get the `(identity, TTL, error)` back?

File: persistence/db_wrapper.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	// always fall through to the database
+	return c.Database.GetIdentity(email)
+}
+
+func (c *CacheWrapper) DeleteIdentity(key string) (bool, error) {
+
+	// TODO - not implemented
+	return false, nil
+}
```
Comment (daiLlew) [inline persistence/db_wrapper.go:61]: Looks good too me.
However **[This comment is not related to this PR just thinking aloud]** I am starting to wonder if we might need to spit the `TokenDB` and `IdentityDB` into separate things? We know we want to cache the token look ups so it makes sense for those methods to wrapped in the cache. However, I am not sure we are going to do that for identities 🤔  We might, but if not this wrapper will just be calling the DB without adding anything.

Comment (mikeAdamss) [reply persistence/db_wrapper.go:61]: That works for me. The IdentityDB is gonna be the source of truth really, so I don't see us ever wanting to cache that. Will iterate with that and see how it looks.

File: cache/nop_cache.go (hunk 2 above)
Comment (mikeAdamss) [reply cache/nop_cache.go:30]: I believe you're correct. Will update.

**Pull request closed without merging**